joblib/numpy_pickle.py

```diff
@@ -384,29 +398,62 @@ def dump(value, filename, compress=0, cache_size=100, protocol=None):
     using compression can significantly slow down loading. In
     addition, compressed files take extra extra memory during
     dump and load.
+
     """
     if compress is True:
-        # By default, if compress is enabled, we want to be using 3 by
-        # default
+        # By default, if compress is enabled, we want to be using 3 by default
         compress = 3
+    elif isinstance(compress, tuple):
+        # a 2-uple was set in compress
+        if len(compress) != 2:
+            raise ValueError(
+                'Compress argument tuple should contain exactly 2 elements: '
+                '(compress method, compress level)')
+        compress_method = compress[0]
+        compress = compress[1]
+
+    if compress is not False and compress not in range(10):
+        # Raising an error if a non valid compress level is given.
+        raise ValueError(
+            'Non valid compress level given: "{0}"'.format(compress))
```

Comment:
A good practice in this kind of error message is to tell the user what are the valid values.